parsons/databases/database/database.py

```diff
@@ -107,6 +115,28 @@ def is_valid_sql_num(self, val):
         except (TypeError, ValueError):
             return False
 
+    def is_sql_bool(self, val):
+        """Check whether val is a valid sql boolean.
+
+        When inserting data into databases, different values can be accepted
+        as boolean types. For excample, ``False``, ``'FALSE'``, ``1``.
+
+        `Args`:
+            val: any
+                The value to check.
+        `Returns`:
+            bool
+                Whether or not the value is a valid sql boolean.
+        """
+        if not self.DO_PARSE_BOOLS:
+            return
+
+        if isinstance(val, bool) or (
+                type(val) in (int, str) and
+                str(val).upper() in self.TRUE_VALS + self.FALSE_VALS):
+            return True
+        return False
+
     def detect_data_type(self, value, cmp_type=None):
```

Comment:
It might be easier to toggle the Boolean parsing on or off if we pass it as an optional argument in this function, instead of the config setting on the base class. Something like: 
`def detect_data_type(self, value, cmp_type=None, do_parse_bools=False):`